Sources_v3/Query/Pagination.swift

```diff
@@ -4,71 +4,75 @@
 
 import Foundation
 
-public typealias Pagination = Set<PaginationOption>
+public extension Int {
+    /// A default channels page size.
+    static let channelsPageSize = 20
+    /// A default messages page size.
+    static let messagesPageSize = 25
+    /// A default users page size.
+    static let usersPageSize = 30
+    /// A default channel members page size.
+    static let channelMembersPageSize = 30
+}
 
-public extension Pagination {
-    var limit: Int? {
-        first(where: { $0.limit != nil })?.limit
+/// Basic pagination with `pageSize` and `offset`.
+/// Used everywhere except `ChannelQuery`. (See `MessagesPagination`)
+public struct Pagination: Encodable, Equatable {
+    /// A page size.
+    let pageSize: Int
+    /// An offset.
+    let offset: Int?
+    
+    private enum CodingKeys: String, CodingKey {
+        case pageSize = "limit"
+        case offset
     }
     
-    var offset: Int? {
-        first(where: { $0.offset != nil })?.offset
+    init(pageSize: Int, offset: Int? = nil) {
+        self.pageSize = pageSize
+        self.offset = offset
     }
     
-    func encode(to encoder: Encoder) throws {
-        try forEach { try $0.encode(to: encoder) }
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(pageSize, forKey: .pageSize)
+        try offset.map { try container.encode($0, forKey: .offset) }
     }
 }
 
-public extension KeyedEncodingContainer {
-    mutating func encode(_ value: Pagination, forKey key: Self.Key) throws {
-        let encoder = superEncoder(forKey: key)
-        try value.forEach { try $0.encode(to: encoder) }
+public struct MessagesPagination: Encodable, Equatable {
```

Comment:
Don't we need `limit` for this too?

Reply:
That is not mentioned in docs and I didn't get confirmation for that 😔